opentelemetry/src/api/trace/mod.rs

```diff
@@ -136,3 +136,14 @@ pub use self::{
     },
     tracer::{SpanBuilder, Tracer},
 };
+
+use thiserror::Error;
+
+/// Errors returned by the trace API.
+#[derive(Error, Debug, PartialEq)]
+#[non_exhaustive]
+pub enum TraceError {
```

Comment:
If we are adding this, should the results below be `Result<(), TraceError>` now?

Reply:
Thought it would be better to include as part of unify error PR. Probably need to change `Other(String)` into `Other(dyn Error + Sync + Send + 'static)` to include all possible errors.

Reply:
Yeah probably a `Box<dyn Error + Send + Sync>` probably makes sense, could have a constructor at some point similar to https://doc.rust-lang.org/std/io/struct.Error.html#method.new